cpplint.py

```diff
@@ -934,9 +934,93 @@ def unicode_escape_decode(x):
 # This is set by --headers flag.
 _hpp_headers = set([])
 
-# {str, bool}: a map from error categories to booleans which indicate if the
-# category should be suppressed for every line.
-_global_error_suppressions = {}
+class ErrorSuppressions:
+  """Class to track all error suppressions for cpplint"""
+
+  class SuppressionRange:
+    """Class to represent a range of line numbers for which an error is suppressed"""
+    def __init__(self, begin=0, end=math.inf):
```

Comment:
nit: Just my personal taste, but code at call sites might be easier to understand if there were no defaults here, but the call sites had to be explicit.

Reply:
Removed